service/frontend/nexus_handler.go

```diff
@@ -121,9 +129,24 @@ func (c *operationContext) matchingRequest(req *nexuspb.Request) *matchingservic
 	}
 }
 
-func (c *operationContext) augmentContext(ctx context.Context) context.Context {
+func (c *operationContext) augmentContext(ctx context.Context, header nexus.Header) context.Context {
 	ctx = metrics.AddMetricsContext(ctx)
-	return interceptor.AddTelemetryContext(ctx, c.metricsHandlerForInterceptors)
+	ctx = interceptor.AddTelemetryContext(ctx, c.metricsHandlerForInterceptors)
+	ctx = interceptor.PopulateCallerInfo(
+		ctx,
+		func() string { return c.namespaceName },
+		func() string { return c.method },
+	)
+	if userAgent, ok := header[headerUserAgent]; ok {
+		parts := strings.Split(userAgent, clientNameVersionDelim)
+		if len(parts) > 1 {
```

Comment:
I think this could be `== 2` since we'll always expect user agents for our SDKs with this format.